src/Position.h

```diff
@@ -45,6 +45,7 @@ struct StateInfo {
 
   // Not copied when making a move (will be recomputed anyhow)
   Key        key;
+  Key        materialKey;
```

Comment:
This is the cause of the bug.  Need to move this up so its above key so that its copied during do_move.  Since do_move doesn't fully recompute it, just calculates a delta for captures.